app/services/slot_builder.rb

```diff
@@ -89,15 +89,20 @@ def slots_for(plage_ouverture_free_times, motif)
 
     def calculate_slots(free_time, motif, plage_ouverture)
       slots = []
-      possible_slot_time = free_time.begin..(free_time.begin + motif.default_duration_in_min.minutes)
-      while possible_slot_time.end <= free_time.end
+      possible_slot_start = free_time.begin
+
+      possible_slot_start += motif.default_duration_in_min.minutes while possible_slot_start <= Time.zone.now
```

Comment:
C'est cette ligne qui fait le fix. Le reste des modifs est un refacto pour que la variable sur laquelle on itère soit la début du créneau, et pas le range du créneau. Ça permet une écriture un peu plus légère des incrémentations.
On pourrait réécrire cette ligne avec la bonne expression pour tout de suite se placer au début possible du créneau, mais j'ai peur qu'on perde trop en lisibilité pour un gain de performance assez léger.
Ça ressemblerait à ça :
```
if possible_slot_start <= Time.zone.now
  possible_slot_start += (Time.zone.now - possible_slot_start) / motif.default_duration_in_min.minutes
end
```
Je suis preneur de retours et d'avis sur ce point :)